lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -763,6 +788,36 @@ void AtenLenTOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenAddTensorOp
+//===----------------------------------------------------------------------===//
+
+void AtenAddTensorOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                                  MLIRContext *context) {
+  patterns.add(+[](AtenAddTensorOp op, PatternRewriter &rewriter) {
+    // The lhs and rhs of the add.tensor op should be 0d tensors for the
+    // canonicalization to be carried out.
+    // `aten.add.tensor(self, other, alpha)` is canonicalized to
+    // `aten.add.int(self, aten.mul.int(other, alpha))`.
+
+    Value lhs = getScalarValue(op.self(), op.getLoc(), rewriter);
+    Value rhs = getScalarValue(op.other(), op.getLoc(), rewriter);
+
+    if (!lhs || !rhs)
+      return failure();
```

Comment:
nit: can you use `rewriter.notifyMatchFailure` with an error message to signal failure? Also for the `failure()` below